shop/src/components/ConfirmationModal.js

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect } from 'react'
+
+import useSetState from 'utils/useSetState'
+import Modal from 'components/Modal'
+
+const AdminConfirmationModal = ({
+  className = 'btn btn-outline-primary',
+  confirmedText = 'Success',
+  confirmText = 'Are you sure?',
+  onConfirm,
+  onSuccess,
+  buttonText,
+  children
+}) => {
+  const [state, setState] = useSetState()
+
+  useEffect(() => {
+    if (state.doConfirm) {
+      onConfirm()
+        .then(() => {
+          setState({ confirmed: true })
+        })
+        .catch((err) => {
+          setState({ error: err.toString() })
+        })
+    }
+  }, [state.doConfirm])
+
+  return (
+    <>
+      <button
+        type="button"
+        className={className}
+        onClick={() => setState({ showModal: true })}
+      >
+        {buttonText}
+      </button>
+      {!state.showModal ? null : (
+        <Modal
+          shouldClose={state.shouldClose}
+          onClose={() => {
+            if (state.confirmed) {
+              onSuccess()
+            } else {
+              setState({}, true)
+            }
+          }}
+        >
```

Comment:
Any thoughts on using Bootstrap model? I know the plan was to switch to them on DApp. But, was wondering if it is also on Dshop pipeline. 

I did add `react-boostrap` package and used the bootstrap Modal in a few places. Can revert that back if we are going to stick with this custom written component

Reply:
The modal in components/Modal is re-written to use the Bootstrap styles, so is an improvement on the one in the DApp. Wanted to avoid react-bootstrap as I think it's a fairly big dependency

Reply:
Got it. Didn't think about the size of react-bootstrap. Will remove it then